### PR TITLE
Drop MySQL 5.7 support in v7

### PR DESCRIPTION
**Pull request by lohart13**
Updating the supported versions for MySQL for `@sequelize/core` version 7 as per sequelize/sequelize#17029

### CODE REVIEW

File: src/pages/releases.mdx [hunk 1]
```diff
@@ -65,7 +65,7 @@ MySQL requires the use of the [mysql2] npm package.
 
 | Sequelize   | [MySQL][mysql] | [mysql2] |
 |-------------|----------------|----------|
-| 7 (alpha)   | ^5.7, ^8.0     | >= 2.3.3 |
+| 7 (alpha)   | >=8.0.19       | >= 3.9.1 |
```
Comment (WikiRik) [inline src/pages/releases.mdx:68]: Not sure about the adapter change, earlier versions should work but we just don't test on it

Comment (lohart13) [reply src/pages/releases.mdx]: Due to the import in the link below, the older version supported is [3.8.0](https://github.com/sidorares/node-mysql2/releases/tag/v3.8.0)

https://github.com/sequelize/sequelize/blob/3b8349586a24e92f15c1f5f4d011b44dcf9235b7/packages/core/src/dialects/mysql/connection-manager.ts#L9

Comment (WikiRik) [reply src/pages/releases.mdx]: Ah yes, that's true. When the user is using TypeScript we'll need 3.8.0

**Review by WikiRik — APPROVED** [review]

**Comment by WikiRik** [comment]
This looks good, but I believe we also reference to 5.7/8.0.13 in other pages (like for the UUID) so I'll check that first before I merge

**Comment by lohart13** [comment]
I've update a couple of links, the only other reference I could find was 

https://github.com/sequelize/website/blob/1fffcbbd517b0ea774336f435aead01895c43fcf/docs/models/defining-models.mdx#L341

File: versioned_docs/version-6.x.x/other-topics/dialect-specific-things.md [hunk 2]
```diff
@@ -276,7 +276,7 @@ Project.findAll({
 
 The `indexHints` option can be used to define index hints. The hint type must be a value from `IndexHints` and the values should reference existing indexes.
 
-Index hints [override the default behavior of the MySQL query optimizer](https://dev.mysql.com/doc/refman/5.7/en/index-hints.html).
+Index hints [override the default behavior of the MySQL query optimizer](https://dev.mysql.com/doc/refman/8.0/en/index-hints.html).
```
Comment (WikiRik) [inline versioned_docs/version-6.x.x/other-topics/dialect-specific-things.md:279]:
```suggestion
Index hints [override the default behavior of the MySQL query optimizer](https://dev.mysql.com/doc/refman/5.7/en/index-hints.html).
```
v6 still supports 5.7

**Comment by WikiRik** [comment]
> I've update a couple of links, the only other reference I could find was

You're right. I was mistakenly thinking we had more references to 5.7.
 
> https://github.com/sequelize/website/blob/1fffcbbd517b0ea774336f435aead01895c43fcf/docs/models/defining-models.mdx#L341

I think we can remove this info, since we do not supports versions under 8.0.13

File: versioned_docs/version-6.x.x/other-topics/dialect-specific-things.md (hunk 2 above)
Comment (lohart13) [reply versioned_docs/version-6.x.x/other-topics/dialect-specific-things.md]: Apologies, one change to many, should be fixed now.

**Comment by lohart13** [comment]
Sure, updated the changes now, LMK if find anything else I need to update.

**Review by WikiRik — APPROVED** [review]

**Pull request merged**